kong/plugins/ldap-auth/access.lua

```diff
@@ -84,14 +86,31 @@ local function load_credential(given_username, given_password, conf)
   return {username = given_username, password = given_password}
 end
 
+
+local function acl_cache_key(conf, username)
+  local cache_key = {
+    "ldap_auth_cache",
+    md5(concat({
+      conf.ldap_host,
+      conf.ldap_port,
+      conf.base_dn,
+      conf.attribute,
+      username,
+      conf.cache_ttl,
+    }, ":"))
+  }
+
+  return concat(cache_key, ":")
+end
+
+
 local function authenticate(conf, given_credentials)
   local given_username, given_password = retrieve_credentials(given_credentials, conf)
   if given_username == nil then
     return false
   end
 
-  local route_id = conf.route_id or conf.api_id
-  local cache_key = "ldap_auth_cache:" .. route_id .. ":" .. given_username
+  local cache_key = acl_cache_key(conf, given_username)
```

Comment:
this is calculated on every request. We better cache the calculated key.

Reply:
Not sure, the calculation is not very intensive. Maybe caching is even slower than calculating MD5 out of small string? Not sure if it is worth considering added complexity?

Reply:
if you leave the username out of the hash, you can create the entire key and store it in a weak table. That would be a single table lookup, followed by the username concatenation.